oracle/src/services/gasPrice.js

```diff
@@ -33,53 +33,18 @@ const foreignBridge = new web3Foreign.eth.Contract(ForeignABI, FOREIGN_BRIDGE_AD
 
 let cachedGasPrice = null
 
-function gasPriceWithinLimits(gasPrice) {
-  if (gasPrice < GAS_PRICE_BOUNDARIES.MIN) {
-    return GAS_PRICE_BOUNDARIES.MIN
-  } else if (gasPrice > GAS_PRICE_BOUNDARIES.MAX) {
-    return GAS_PRICE_BOUNDARIES.MAX
-  } else {
-    return gasPrice
-  }
-}
-
-function normalizeGasPrice(oracleGasPrice, factor) {
-  const gasPriceGwei = oracleGasPrice * factor
-  const gasPrice = gasPriceWithinLimits(gasPriceGwei)
-  return Web3Utils.toWei(gasPrice.toFixed(2).toString(), 'gwei')
-}
-
-async function fetchGasPriceFromOracle(oracleUrl, speedType, factor) {
-  const response = await fetch(oracleUrl)
-  const json = await response.json()
-  const oracleGasPrice = json[speedType]
-  if (!oracleGasPrice) {
-    throw new Error(`Response from Oracle didn't include gas price for ${speedType} type.`)
-  }
-
-  return normalizeGasPrice(oracleGasPrice, factor)
-}
-
-async function fetchGasPrice({ bridgeContract, oracleFn }) {
-  let gasPrice = null
-  try {
-    gasPrice = await oracleFn()
-    logger.debug({ gasPrice }, 'Gas price updated using the oracle')
-  } catch (e) {
-    logger.error(`Gas Price API is not available. ${e.message}`)
+let fetchGasPriceInterval = null
 
-    try {
-      gasPrice = await bridgeContract.methods.gasPrice().call()
-      logger.debug({ gasPrice }, 'Gas price updated using the contracts')
-    } catch (e) {
-      logger.error(`There was a problem getting the gas price from the contract. ${e.message}`)
-    }
-  }
-  return gasPrice
+const fetchGasPrice = async (speedType, factor, bridgeContract, oracleFetchFn) => {
+  const contractOptions = { logger }
+  const oracleOptions = { speedType, factor, limits: GAS_PRICE_BOUNDARIES, logger }
+  cachedGasPrice =
+    (await gasPriceFromContract(bridgeContract, contractOptions)) ||
+    (await gasPriceFromOracle(oracleFetchFn, oracleOptions)) ||
```

Comment:
I think the order is not correct, we should get gas price from the Oracle first, then from the contract, then fallback value.

So this should be re-defined as:
```javascript
cachedGasPrice =
  (await gasPriceFromOracle(oracleFetchFn, oracleOptions)) ||
  (await gasPriceFromContract(bridgeContract, contractOptions)) ||
  cachedGasPrice
```

Reply:
You're right, it's a mistake, the original order was to get from oracle/supplier first.

Updated.